### PR TITLE
[FEATURE] Inline Blame View

### PR DESCRIPTION
**Pull request by f4n0**

### CODE REVIEW

**Comment by f4n0** [comment]
#94

File: src/git/extension.ts [hunk 1]
```diff
@@ -29,12 +33,19 @@ type ActionableMessageItem = MessageItem & {
...
     private readonly disposable: Disposable;
     private readonly blame: Blamer;
     private readonly view: StatusBarView;
     private readonly headWatcher: HeadWatch;
 
+    /**
+     * decorationCharPosition set to 1024 because normally lines are not that big
+     */
+    private readonly decorationCharPosition: number = 1024;
+    private decorationType: TextEditorDecorationType = window.createTextEditorDecorationType({});
```
Comment (Sertion) [inline src/git/extension.ts:47]: Please move the initialization of the `decorationType` to the constructor to keep in more in line with the other private variables.

Comment (Sertion) [reply src/git/extension.ts:47]: Also please add it to the `dispose` function below.

File: src/git/extension.ts [hunk 2]
```diff
@@ -167,6 +178,36 @@ export class Extension {
...
+                //add new decoration
+                textEditor.setDecorations?.(
+                    this.decorationType,
+                    [
+                        {
+                            renderOptions: {
+                                after: {
+                                    contentText: decorationText,
+                                    margin: `0 0 0 ${margin}rem`,
+                                    color: new ThemeColor("gitblame.inlineBlameColor"),
```
Comment (Sertion) [inline src/git/extension.ts:198]: I think this misinterprets how the `ThemeColor` works. The `id` should be [from this list](https://code.visualstudio.com/api/references/theme-color). I would personally chose `editorCodeLens.foreground`.

If we do that we also don't need a setting for it and that can be removed.

**Review by Sertion — COMMENTED** [review]
Hey f4n0. Thank you for the PR!

Let's continue the discussion about changes here rather than the issue.

Please have a look at the lint run and I have a few comments amongst the changes.

**Comment by f4n0** [comment]
Done, I've modified it as you asked!
and also I fixed some lint errors

**Comment by Sertion** [comment]
Thank you for the quick turn-around. There are a few auto-fixable lint errors left and then I'll merge.

**Comment by Sertion** [comment]
That test failure is due to an issue in the test itself. I'll merge this now. I'll also add a setting to disable this for those who want to keep the old behavior.

**Comment by Sertion** [comment]
Thank you!

**Pull request merged**